src/test/java/com/squareup/javapoet/MethodSpecTest.java

```diff
@@ -83,74 +109,95 @@ private TypeElement getElement(Class<?> clazz) {
-    ExecutableElement methodElement = getOnlyElement(methodsIn(classElement.getEnclosedElements()));
+    TypeElement classElement = getElement(HasAnnotation.class);
+    DeclaredType classType = (DeclaredType) classElement.asType();
+    ExecutableElement exec = getOnlyElement(methodsIn(classElement.getEnclosedElements()));
 
-    MethodSpec method = MethodSpec.overriding(methodElement).build();
+    MethodSpec method = MethodSpec.overriding(exec, classType, elements, types).build();
     assertThat(method.toString()).isEqualTo(""
         + "@java.lang.Override\n"
-        + "public java.lang.String toString() {"
-        + "}");
+        + "public java.lang.String toString() {\n"
+        + "}\n");
+  }
+
+
+  @Test public void overrideExtendsOthersWorksWithActualTypeParameters() {
+    TypeElement classElement = getElement(ExtendsOthers.class);
+    DeclaredType classType = (DeclaredType) classElement.asType();
+    List<ExecutableElement> methods = methodsIn(elements.getAllMembers(classElement));
+
+    ExecutableElement exec = findFirst(methods, "call");
+    MethodSpec method = MethodSpec.overriding(exec, classType, elements, types).build();
+    assertThat(method.toString()).isEqualTo(""
+        + "@java.lang.Override\n"
+        + "public java.lang.Integer call() throws java.lang.Exception {\n"
+        + "}\n");
+
+    exec = findFirst(methods, "compareTo");
+    method = MethodSpec.overriding(exec, classType, elements, types).build();
+    assertThat(method.toString()).isEqualTo(""
+        + "@java.lang.Override\n"
+        + "public int compareTo(java.lang.Long arg0) {\n"
+        + "}\n");
   }
 
   @Test public void overrideInvalidModifiers() {
     ExecutableElement method = mock(ExecutableElement.class);
     when(method.getModifiers()).thenReturn(ImmutableSet.of(Modifier.FINAL));
+    Element element = mock(Element.class);
+    when(element.asType()).thenReturn(mock(DeclaredType.class));
+    when(method.getEnclosingElement()).thenReturn(element);
     try {
-      MethodSpec.overriding(method);
+      MethodSpec.overriding(method, elements, types);
       fail();
     } catch (IllegalArgumentException expected) {
       assertThat(expected).hasMessage("cannot override method with modifiers: [final]");
     }
     when(method.getModifiers()).thenReturn(ImmutableSet.of(Modifier.PRIVATE));
     try {
-      MethodSpec.overriding(method);
+      MethodSpec.overriding(method, elements, types);
       fail();
     } catch (IllegalArgumentException expected) {
       assertThat(expected).hasMessage("cannot override method with modifiers: [private]");
     }
     when(method.getModifiers()).thenReturn(ImmutableSet.of(Modifier.STATIC));
     try {
-      MethodSpec.overriding(method);
+      MethodSpec.overriding(method, elements, types);
       fail();
     } catch (IllegalArgumentException expected) {
       assertThat(expected).hasMessage("cannot override method with modifiers: [static]");
     }
   }
+
```

Comment:
Empty lines removed. Although, the "code paragraphs" show the outline of the algorythm better, imho.